dataset/test/groupby_test.cpp

```diff
@@ -371,6 +370,37 @@ TEST_F(GroupbyWithBinsTest, two_bin) {
   EXPECT_EQ(groups.mean(Dim::X).slice({Dim::Z, 1}), mean(group1, Dim::X));
 }
 
+TEST_F(GroupbyWithBinsTest, dataset_variable) {
```

Comment:
Can we somehow avoid writting new "low-level" test code, e.g., by testing the new overload against the old one, and using the same test data as some tests above?

Maybe testing indirectly via `sum` is also not ideal? Can we directly compare the results of `groupby`? E.g., comparing the results of `groupby(data, key_label, bins)` and `groupby(data, key_variable, bins)`? There is no comparison operator, but maybe this is worth adding?

Reply:
Indeed.
Comparing `groupby.key()`s is probably enough, since this is where the implementations differ.
Added.